libraries/rush-lib/src/logic/ProjectWatcher.ts

```diff
@@ -180,8 +210,26 @@ export class ProjectWatcher {
                 return;
               }
 
+              // Since there are multiple async ticks since the projects were enumerated in _computeChanged,
+              // more could have been added in the interaval. Check and debounce.
+              for (const project of this._forceChangedProjects.keys()) {
+                if (!result.changedProjects.has(project)) {
+                  this._setStatus(`More invalidations occurred, aborting.`);
+                  timeout = setTimeout(resolveIfChanged, debounceMs);
+                  return;
+                }
+              }
+
               this._commitChanges(result.state);
 
+              const hasForcedChanges: boolean = this._forceChangedProjects.size > 0;
+              if (hasForcedChanges) {
+                this._setStatus(
+                  `Projects were invalidated: ${Array.from(new Set(this._forceChangedProjects.values()))}`
```

Comment:
You don't want to explicitly format this array?

Reply:
Huh, thought I had a `.join` on there. Good catch.